### PR TITLE
Use `lipgloss.Width` to calculate `DisplayWidth`

### PR DESCRIPTION
**Pull request by maaslalani**
Fixes https://github.com/cli/cli/issues/9018

Uses `lipgloss.Width` to calculate `DisplayWidth` which correctly aligns emojis.

### CODE REVIEW

**Comment by maaslalani** [comment]
image

File: pkg/text/text.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func Indent(s, indent string) string {
 
 // DisplayWidth calculates what the rendered width of string s will be.
 func DisplayWidth(s string) int {
-	return ansi.PrintableRuneWidth(s)
+	return uniseg.StringWidth(s)
```
Comment (williammartin) [inline pkg/text/text.go]: **Change Requested**

Can you add the emoji from https://github.com/cli/cli/issues/9018 into the `TestDisplayWidth` test cases?

**Review by williammartin — CHANGES_REQUESTED** [review]

**Comment by maaslalani** [comment]
@williammartin Had to switch to `lipgloss.Width` to account for ANSI sequences since `uniseg` doesn't strip ANSI before calculating width (one of the test cases included ANSI sequences), but `lipgloss` does then uses `uniseg.StringWidth` under the hood.

Added the multi-byte emoji test case as well!

File: go.mod [hunk 2]
```diff
@@ -17,7 +18,7 @@ require (
 	github.com/muesli/termenv v0.15.2
 	github.com/stretchr/testify v1.7.0
 	github.com/thlib/go-timezone-local v0.0.0-20210907160436-ef149e42d28e
-	golang.org/x/sys v0.18.0
+	golang.org/x/sys v0.19.0
```
Comment (maaslalani) [inline go.mod:21]: Let me know if you don't want to bump this.

**Comment by williammartin** [comment]
Thanks. If we're expecting `DisplayWidth` to handle ANSI escape sequences then let's add a test for that as well.

I suppose for someone out there this could be a breaking change but I think it's edge case enough that I'd rather just accept it and then people could fix any hacky workarounds they might have had.

**Comment by maaslalani** [comment]
Yep, there is already a test case for that! The tests were failing when I simply had `uniseg.StringWidth` because of the ANSI test case. Which is why I needed to swap out for lipgloss to pass all tests cases.

**Comment by maaslalani** [comment]
Test case for ignoring ANSI sequences when calculating width is here: https://github.com/cli/go-gh/blob/trunk/pkg/text/text_test.go#L282-L287 (fails when _just_ using `uniseg.StringWidth`, passes when using `lipgloss.Width`)

**Comment by williammartin** [comment]
Well, that's testing the public API of `PadRight` but it's just an implementation detail that `PadRight` uses `DisplayWidth`. Refactoring `PadRight` would then lose all our confidence that `DisplayWidth` handles ANSI codes. I want to both ensure that `DisplayWidth` enforces this behaviour, and I want it to be communicated to future maintainers in the tests because I had no idea that it was expected when you first opened the PR.

**Comment by maaslalani** [comment]
Ah you're totally right, there is also a test case directly for color codes in `DisplayWidth` as well:

https://github.com/cli/go-gh/blob/5840c447a66255f2822dfc9be315be0f9ff1d6d2/pkg/text/text_test.go#L360-L363

Happy to add more test cases though if you want / prefer!

**Comment by williammartin** [comment]
Haha sorry I should have also looked closer. I think I got red herring-ed by your link because I assumed if it existed already you would definitely have linked it 😅 

I can confirm that test fails just using `uniseg.StringWidth`:

```
--- FAIL: TestDisplayWidth (0.00s)
    --- FAIL: TestDisplayWidth/color_codes (0.00s)
        /Users/williammartin/workspace/go-gh/pkg/text/text_test.go:388: 
            	Error Trace:	text_test.go:388
            	Error:      	Not equal: 
            	            	expected: 3
            	            	actual  : 12
            	Test:       	TestDisplayWidth/color_codes
```

No need for extra tests, this looks great to me, once I pull it into `cli/cli` and play with it, cheers.

**Comment by maaslalani** [comment]
> I think I got red herring-ed by your link because I assumed if it existed already you would definitely have linked it 😅

Haha all good! I kind of missed that test case the first time as well and definitely should have linked to that!

**Comment by williammartin** [comment]
### Before

image

(wow such codepoints: https://ardislu.dev/biggest-unicode-grapheme-cluster)

### After

image

**Review by williammartin — APPROVED** [review]
I pulled this into `cli/cli` and it passed all the tests.

LGTM thanks a lot.

**Review by Abeabenut — COMMENTED** [review]
https://github.com/cli/cli/releases/tag/v2.48.0
